dart/lib/src/sentry_tracer.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:meta/meta.dart';
+
+import '../sentry.dart';
+
+@internal
+class SentryTracer extends ISentrySpan {
+  final Hub _hub;
+  late String name;
+
+  // missing waitForChildren
+
+  late final SentrySpan _rootSpan;
+  final List<SentrySpan> _children = [];
+  final Map<String, String> _extra = {};
+
+  SentryTracer(SentryTransactionContext transactionContext, this._hub) {
+    _rootSpan = SentrySpan(
+      this,
+      transactionContext,
+      _hub,
+      sampled: transactionContext.sampled,
+    );
+    name = transactionContext.name;
+  }
+
+  @override
+  Future<void> finish({SpanStatus? status}) async {
+    await _rootSpan.finish(status: status);
+
+    // finish unfinished spans otherwise transaction gets dropped
+    for (final span in _children) {
+      if (!span.finished) {
+        await span.finish(status: SpanStatus.deadlineExceeded());
+      }
+    }
+
+    // remove from scope
+    _hub.configureScope((scope) {
+      if (scope.span == this) {
+        scope.span = null;
+      }
+    });
+
+    final transaction = SentryTransaction(this);
+    await _hub.captureTransaction(transaction);
```

Comment:
I would like to add the transaction to the breadcrumb, that requires to have the eventid and that would only be possible if the finish method would return the sentryId or the event. is that a good idea? if yes I can create a MR, thx!

Reply:
Events are automatically connected to the transaction. What would this improve on? 
You should probably create an issue to better describe the use case.